### ISSUE TITLE
Why binned likelihood use cdf instead of pdf?

### ISSUE THREAD

**Issue by DingXuefeng** [scikit-hep/iminuit#473]
To perform binned likelihood fit, the likelihood is calculated on each bin using the pdf, not the cdf. There is numerical problems for subtracting two tiny numbers, if the tail contains very little events. In `src/imiuit/cost.py`, it is written

```python
prob = np.diff(self.cdf(self.xe, *args))
```

Why not use pdf directly?

**Comment by HDembinski** [comment]
Using the cdf is correct for bins of finite size, using the pdf is an approximation and would be correct only for bin widths that go to zero.

**Comment by DingXuefeng** [comment]
If you read more carefully what I write, I am not talking about taking the pdf value at the bin center. Let me restate the problem. Because you called `np.diff` function, there is numerical preicsion problem. Considering our pdf has 4 orders of magnitude difference between low energies and high energies, this subtraction will arbitrary degrade the double precision to single precision.

Instead of providing the cdf, the function should take "integration of pdf inside the bin", or "cdf_diff" directly, rather than calculating it with np.diff and cdf inside iminuit.

**Comment by HDembinski** [comment]
I assume you mean that the CDF is four magnitudes larger, not the pdf. Four orders should not be an issue for double precision.

Is this a real problem you are facing or a hypothetical problem?

**Comment by DingXuefeng** [comment]
This is more about a discussion of the design of the interface. Since python is flexible, there are no real problems. There are always workarounds. For example, a customized cost function can be used, as we did in our production analysis. I believe it's more natural to request direct input rather than manipulate it inside, but It seems the current design is trying to force users to avoid using the pdf value at the center.

As for potential bias, I did not redesign our package with `iminuit`. Since I designed our continuous integration pipeline, I'm aware that this would break the test.
